### PR TITLE
Support parametric arguments to `FUNCTION`  for ClickHouse dialect

### LINKED ISSUE #1301: Support Clickhouse parametric aggregation functions
https://clickhouse.com/docs/en/sql-reference/aggregate-functions/parametric-functions

```sql
quantile(0.9)(col)
```

### PR DESCRIPTION
**Pull request by git-hulk**
This closes #1301

### CODE REVIEW

File: src/ast/mod.rs [hunk 1]
```diff
@@ -4706,6 +4706,9 @@ impl fmt::Display for CloseCursor {
 #[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
 pub struct Function {
     pub name: ObjectName,
+    /// The parameters to the function, including any options specified within the
+    /// delimiting parentheses.
```
Comment (iffyio) [inline src/ast/mod.rs:4710]: Can we include a link to the clickhouse docs?

File: src/parser/mod.rs [hunk 2]
```diff
@@ -1298,7 +1301,14 @@ impl<'a> Parser<'a> {
             }));
         }
 
-        let args = self.parse_function_argument_list()?;
+        let mut args = self.parse_function_argument_list()?;
+        let mut parameters = FunctionArguments::None;
+        // ClickHouse aggregation support parametric functions like `quantile(0.5)(x)`
+        // which (0.5) is a parameter to the function.
+        if dialect_of!(self is ClickHouseDialect) && self.consume_token(&Token::LParen) {
```
Comment (iffyio) [inline src/parser/mod.rs:1308]:
```suggestion
        if dialect_of!(self is ClickHouseDialect | GenericDialect) && self.consume_token(&Token::LParen) {
```
we can include generic dialect if no conflicts

File: src/parser/mod.rs [hunk 3]
```diff
@@ -1298,7 +1301,14 @@ impl<'a> Parser<'a> {
             }));
         }
 
-        let args = self.parse_function_argument_list()?;
+        let mut args = self.parse_function_argument_list()?;
+        let mut parameters = FunctionArguments::None;
+        // ClickHouse aggregation support parametric functions like `quantile(0.5)(x)`
```
Comment (iffyio) [inline src/parser/mod.rs:1306]:
```suggestion
        // ClickHouse aggregations support parametric functions like `quantile(0.5)(x)`
```

File: tests/sqlparser_clickhouse.rs [hunk 4]
```diff
@@ -553,6 +554,11 @@ fn parse_select_star_except() {
     clickhouse().verified_stmt("SELECT * EXCEPT (prev_status) FROM anomalies");
 }
 
+#[test]
+fn parse_select_parametric_function() {
+    clickhouse().verified_stmt("SELECT quantile(0.5)(x) FROM t");
+}
```
Comment (iffyio) [inline tests/sqlparser_clickhouse.rs:560]: Could we extend the test asserting that the parameter and arguments lists do indeed show up in the `parameters` and `args` fields respectively (i.e not vice versa for example). Also we can include multiple items in both lists e.g `quantile(0.5,0.6)(x,y)`

File: src/ast/mod.rs (hunk 1 above)
Comment (git-hulk) [reply src/ast/mod.rs:4710]: @iffyio Done

**Comment by git-hulk** [comment]
@iffyio Thanks for your review, all comments are resolved now.

**Review by iffyio — APPROVED** [review]
LGTM! cc @alamb

**Review by alamb — APPROVED** [review]
Thanks @git-hulk and @iffyio

File: src/parser/mod.rs [hunk 5]
```diff
@@ -8100,6 +8115,17 @@ impl<'a> Parser<'a> {
             );
         };
 
+        self.parse_remaining_set_exprs(expr, precedence)
+    }
+
+    /// Parse any extra set expressions that may be present in a query body
+    ///
```
Comment (alamb) [inline src/parser/mod.rs:8122]: I also pushed a commit to split up this function in e2ec2a6 to get the deeply nested tests to pass

**Comment by alamb** [comment]
Thanks again @git-hulk  and @iffyio

**Pull request merged**